File: src/nunja/stock/tests/test_render_mold.js
'use strict';

var loader = require('nunja/stock/tests/loader');

window.mocha.setup('bdd');


loader.generate_test(
    'nunja.stock.molds/breadcrumb rendering from JSON module',
    'nunja.stock.molds/breadcrumb',
    'nunja/stock/tests/breadcrumb_examples'
);


loader.generate_test(
    'nunja.stock.molds/grid rendering from JSON module',
    'nunja.stock.molds/grid',
    'nunja/stock/tests/grid_examples'
);


loader.generate_test(
    'nunja.stock.molds/navgrid rendering from JSON module',
    'nunja.stock.molds/navgrid',
    'nunja/stock/tests/navgrid_examples'
);


loader.generate_test(
    'nunja.stock.molds/model rendering from JSON module',
    'nunja.stock.molds/model',
    'nunja/stock/tests/model_examples'
);


loader.generate_test(
    'nunja.stock.molds/model navgrid rendering from JSON module',
    'nunja.stock.molds/model',
    'nunja/stock/tests/model_navgrid_examples'
);

File: src/nunja/stock/tests/test_render_mold.py
# -*- coding: utf-8 -*-
from nunja.stock.testing.loader import generate_test


GridRenderTestCase = generate_test(
    name='GridRenderTestCase',
    mold_id='nunja.stock.molds/grid',
    test_module_ns='nunja.stock.tests',
    data_module='grid_examples',
)


NavGridRenderTestCase = generate_test(
    name='NavGridRenderTestCase',
    mold_id='nunja.stock.molds/navgrid',
    test_module_ns='nunja.stock.tests',
    data_module='navgrid_examples',
)


ModelRenderTestCase = generate_test(
    name='ModelRenderTestCase',
    mold_id='nunja.stock.molds/model',
    test_module_ns='nunja.stock.tests',
    data_module='model_examples',
)


ModelNavGridRenderTestCase = generate_test(
    name='ModelNavGridRenderTestCase',
    mold_id='nunja.stock.molds/model',
    test_module_ns='nunja.stock.tests',
    data_module='model_navgrid_examples',
)
